api/server.go

```diff
@@ -69,9 +69,6 @@ func Serve(ctx context.Context, serveAddr string, db idb.IndexerDb, fetcherError
 	generated.RegisterHandlers(e, &api, middleware...)
 	common.RegisterHandlers(e, &api)
 
-	if ctx == nil {
-		ctx = context.Background()
-	}
 	getctx := func(l net.Listener) context.Context {
 		return ctx
 	}
```

Comment:
The nil check is required to avoid violating BaseContext constraints (and a panic below), from the docs:
```
	// If BaseContext is nil, the default is context.Background().
	// If non-nil, it must return a non-nil context.
```

Reply:
We can just assume that `Serve()` accepts non-nil context? It's a common assumption, I've never seen `nil` passed as a context parameter.

Reply:
No, I don't think we should panic when it's entirely avoidable. Especially when it's already handled correctly.

Reply:
Ok, I added it back.